rclcpp/test/rclcpp/test_executor.cpp

```diff
@@ -122,14 +122,14 @@ TEST_F(TestExecutor, constructor_bad_guard_condition_init) {
   auto mock = mocking_utils::patch_and_return(
     "lib:rclcpp", rcl_guard_condition_init, RCL_RET_ERROR);
   EXPECT_THROW(
-    new DummyExecutor(),
+    static_cast<void>(std::make_unique<DummyExecutor>()),
     rclcpp::exceptions::RCLError);
 }
 
 TEST_F(TestExecutor, constructor_bad_wait_set_init) {
   auto mock = mocking_utils::patch_and_return("lib:rclcpp", rcl_wait_set_init, RCL_RET_ERROR);
   RCLCPP_EXPECT_THROW_EQ(
-    new DummyExecutor(),
+    static_cast<void>(std::make_unique<DummyExecutor>()),
```

Comment:
I don't think we need to do anything with this, but why is this a pointer in the first place? Couldn't it be on the stack? Unlike some entities in rclcpp, the executor need not be a smart pointer.

Reply:
It would probably end up longer anyways, because we'd need to "use" the variable we create to avoid unused variable warnings.

Reply:
I honestly don't remember why I wrote it this way originally.  I guess I could move it to the stack, but like you say it would probably be longer.  So I'll leave it for now.